src/sn_utils.c

```diff
@@ -204,23 +204,25 @@ static int try_broadcast(n2n_sn_t * sss,
 
 
 /** Initialise the supernode structure */
-int sn_init(n2n_sn_t *sss)
-{
+int sn_init(n2n_sn_t *sss) {
 #ifdef WIN32
-    initWin32();
+	initWin32();
 #endif
 
-    pearson_hash_init();
+	pearson_hash_init();
 
-    memset(sss, 0, sizeof(n2n_sn_t));
+	memset(sss, 0, sizeof(n2n_sn_t));
 
-    sss->daemon = 1; /* By defult run as a daemon. */
-    sss->lport = N2N_SN_LPORT_DEFAULT;
-    sss->mport = N2N_SN_MGMT_PORT;
-    sss->sock = -1;
-    sss->mgmt_sock = -1;
+	sss->daemon = 1; /* By defult run as a daemon. */
+	sss->lport = N2N_SN_LPORT_DEFAULT;
+	sss->mport = N2N_SN_MGMT_PORT;
+	sss->sock = -1;
+	sss->mgmt_sock = -1;
+	sss->dhcp_addr.net_addr = inet_addr(N2N_SN_DHCP_NET_ADDR_DEFAULT);
+	sss->dhcp_addr.net_addr = ntohl(sss->dhcp_addr.net_addr);
+	sss->dhcp_addr.net_bitlen = N2N_SN_DHCP_NET_BIT_DEFAULT;
```

Comment:
New. Logic (where to get – new or not new – sub-network and corresponding edge IP address) needs to be re-thought.